geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/ShowMetricsCommand.java

```diff
@@ -85,8 +85,8 @@ public Result showMetrics(
       if (regionName != null && !regionName.isEmpty()) {
 
         if (StringUtils.isNotBlank(cacheServerPortString)) {
-          return ResultBuilder
-              .createUserErrorResult(CliStrings.SHOW_METRICS__CANNOT__USE__CACHESERVERPORT);
+          return ResultBuilder.createUserErrorResult(
```

Comment:
It's better to do these validation of these sort in the client VM rather than having it executed on the manager. You can see ExecuteFunctionCommand as an example.